src/app.jsx

```diff
@@ -55,38 +59,48 @@ export const Application = () => {
     useEffect(() => {
         if (path === undefined)
             return;
-        setFiles([]);
-        const currentPath = path.slice(0, pathIndex).join("/");
-        const channel = cockpit.channel({
-            payload: "fslist1",
-            path: `/${currentPath}`,
-            superuser: "try",
-            watch: true,
-        });
 
-        channel.addEventListener("message", (ev, data) => {
-            const item = JSON.parse(data);
-            if (item.event === 'present' && item.path[0] !== ".")
-                setFiles(f => [...f, { name: item.path, type: item.type, modified: item.modified }]);
-            else if (item.event === 'deleted') {
-                const deleted_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                setFiles(f => f.filter((res) => { return res.name !== deleted_name }));
-            } else if (item.event === 'created') {
-                const created_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                const date_modified = item.tag.slice(item.tag.indexOf("-") + 1, item.tag.lastIndexOf("."));
-                if (created_name[0] !== ".") {
-                    setFiles(f => [...f, { name: created_name, type: item.type, modified: date_modified }]);
+        const getFsList = () => {
+            if (channel.current !== null)
+                channel.current.close();
+
+            const currentPath = path.slice(0, pathIndex).join("/");
+            channel.current = cockpit.channel({
+                payload: "fslist1",
+                path: `/${currentPath}`,
+                superuser: "try",
+                watch: true,
+            });
+
+            const files = [];
+            channel.current.addEventListener("message", (ev, data) => {
+                const item = JSON.parse(data);
+                if (item.event === "present") {
+                    if (item.path[0] !== ".") {
+                        // TODO: Show hidden files if the user has enabled that option
+                        files.push({ ...item, name: item.path });
+                    }
+                } else {
+                    const name = item.path.slice(item.path.lastIndexOf("/") + 1);
+                    if (item.event === 'deleted') {
+                        setFiles(f => f.filter(res => res.name !== name));
```

Comment:
That looks strange here. But I figure that's for the "watch" portion only. We have to assume that the initial read only does "present" events.

Reply:
Yes I did not see present coming on watching, and also the docs are somehow clear the present comes first and the the watch mode lists events that fswatch1 would emit.

Reply:
https://github.com/cockpit-project/cockpit/blob/main/doc/protocol.md#payload-fslist1